pkg/cloudprovider/aws/suite_test.go

```diff
@@ -214,6 +214,37 @@ var _ = Describe("Allocation", func() {
 				}
 				Expect(nodeNames.Len()).To(Equal(2))
 			})
+			It("should launch pods with Nvidia and Neuron resources on different instances", func() {
```

Comment:
I remember asking something similar, but when can we move these tests to generic code? Can we at least cut an OE issue?

Reply:
We could do this at the same type as making resource handling generic, that PR I've got will need to be updated after this merge.  I think it's still waiting on a consensus regarding sorting instance types and 'preferred pod instance type compatibility' (e.g. don't schedule a non-GPU workload on a GPU instance type just because there are plenty of GPU instance types available).  My preference is a comparison method on cloud provider for the first, and just letting cloud providers do the second part internally, which they could do now.

Wrote up an issue here: https://github.com/aws/karpenter/issues/1516